### PR TITLE
Documentation copy and style edits Delta Lake

### PR DESCRIPTION
**Pull request by tlblessing**
## Description

Edits to make corrections for:

- Improper grammar
- Typos
- Improper capitalization
- Incorrect punctuation, such as comma usage 
- Add adjectives to make subjects clear
- Formatting for default values
- Line wrapping to 80 char where possible per conventions

Noticed these last year while working on fixing the description of the delta.domain-compaction-threshold export config property.

## Additional context and related issues

- https://github.com/trinodb/trino/pull/12965
- https://github.com/trinodb/trino/pull/13039

## Release notes

(x) This is not user-visible or docs only and no release notes are required.
( ) Release notes are required, please propose a release note for me.
( ) Release notes are required, with the following suggested text:

### CODE REVIEW

File: docs/src/main/sphinx/connector/delta-lake.rst [hunk 1]
```diff
@@ -452,12 +458,12 @@ Flush metadata cache
...
 
-  Flush all metadata caches.
+  Flushes all metadata caches.
 
 * ``system.flush_metadata_cache(schema_name => ..., table_name => ...)``
 
-  Flush metadata caches entries connected with selected table.
-  Procedure requires named parameters to be passed
+  Flushes metadata cache entries connected with selected table.
+  Procedure requires passing named parameters.
```
Comment (tlblessing) [inline docs/src/main/sphinx/connector/delta-lake.rst:466]: Rewrote to active rather than passive "to be..."

File: docs/src/main/sphinx/connector/delta-lake.rst [hunk 2]
```diff
@@ -735,10 +739,10 @@ following sections:
 Table statistics
 ^^^^^^^^^^^^^^^^
 
-You can use :doc:`/sql/analyze` statements in Trino to populate the table
-statistics in Delta Lake. Data size and number of distinct values (NDV)
-statistics are supported, while Minimum value, maximum value, and null value
-count statistics are not supported. The :doc:`cost-based optimizer
+Use :doc:`/sql/analyze` statements in Trino to populate the table statistics in
+Delta Lake. Data size and number of distinct values (NDV) statistics are
+supported; whereas minimum value, maximum value, and null value count statistics
```
Comment (tlblessing) [inline docs/src/main/sphinx/connector/delta-lake.rst:744]: image

 while vs whereas grammar

File: docs/src/main/sphinx/connector/delta-lake.rst [hunk 3]
```diff
@@ -479,7 +485,7 @@ Write operations are supported for tables stored on the following systems:
 
   Writes to :doc:`Amazon S3 <hive-s3>` and S3-compatible storage must be enabled
   with the ``delta.enable-non-concurrent-writes`` property. Writes to S3 can
-  safely be made from multiple Trino clusters, however write collisions are not
+  safely be made from multiple Trino clusters; however, write collisions are not
```
Comment (tlblessing) [inline docs/src/main/sphinx/connector/delta-lake.rst:488]: image

File: docs/src/main/sphinx/connector/delta-lake.rst [hunk 4]
```diff
@@ -912,26 +916,28 @@ connector.
...
-        can also use the corresponding catalog session property
+        assigned to a worker after ``max-initial-splits`` have been processed.
+        You can also use the corresponding catalog session property
         ``<catalog-name>.max_split_size``.
       - ``64MB``
     * - ``delta.minimum-assigned-split-weight``
-      - A decimal value in the range (0, 1] used as a minimum for weights assigned to each split. A low value may improve performance
-        on tables with small files. A higher value may improve performance for queries with highly skewed aggregations or joins.
-      - 0.05
+      - A decimal value in the range (0, 1] used as a minimum for weights
```
Comment (tlblessing) [inline docs/src/main/sphinx/connector/delta-lake.rst:924]: Reviewers, made no changes here, just wrapped lines to 80 char. Is (0, 1] correct?

File: docs/src/main/sphinx/connector/delta-lake.rst [hunk 5]
```diff
@@ -6,23 +6,25 @@ Delta Lake connector
...
-transaction log, and thus detect when external systems change data.
+<https://docs.databricks.com/delta/index.html>`_. The connector can natively
+read the Delta transaction log and thus detect when external systems change
+data.
 
 Requirements
 ------------
 
-To connect to Databricks Delta Lake, you need:
+To connect to Databricks Delta Lake, you need the following:
```
Comment (mosabua) [inline docs/src/main/sphinx/connector/delta-lake.rst:18]: Lets not do this in this PR .. this is the same in all other connectors .. so if we want to change it .. do it in a separate PR for all connectors maybe .. but dont worry for now

File: docs/src/main/sphinx/connector/delta-lake.rst [hunk 6]
```diff
@@ -6,23 +6,25 @@ Delta Lake connector
...
-* Tables written by Databricks Runtime 7.3 LTS, 9.1 LTS, 10.4 LTS and 11.3 LTS are supported.
-* Deployments using AWS, HDFS, Azure Storage, and Google Cloud Storage (GCS) are
-  fully supported.
+* Tables written by Databricks Runtime 7.3 LTS, 9.1 LTS, 10.4 LTS, and 11.3 LTS
+  versions are supported.
 * Network access from the coordinator and workers to the Delta Lake storage.
 * Access to the Hive metastore service (HMS) of Delta Lake or a separate HMS.
 * Network access to the HMS from the coordinator and workers. Port 9083 is the
   default port for the Thrift protocol used by the HMS.
+* Deployments using AWS, HDFS, Azure Storage, and Google Cloud Storage (GCS) are
```
Comment (mosabua) [inline docs/src/main/sphinx/connector/delta-lake.rst:26]: Why the reorder?

File: docs/src/main/sphinx/connector/delta-lake.rst [hunk 7]
```diff
@@ -388,17 +390,18 @@ to register them::
     location = '...'
   )
 
-Columns listed in the DDL, such as ``dummy`` in the preceeding example, are
-ignored. The table schema is read from the transaction log, instead. If the
+Columns listed in the DDL, such as ``dummy`` in the proceeding example, are
```
Comment (mosabua) [inline docs/src/main/sphinx/connector/delta-lake.rst:393]: preceding

File: docs/src/main/sphinx/connector/delta-lake.rst (hunk 3 above)
Comment (mosabua) [reply docs/src/main/sphinx/connector/delta-lake.rst:488]: No need for such references.

**Review by mosabua — CHANGES_REQUESTED** [review]
Good improvements. Just needs a few small updates and then ready to go.

File: docs/src/main/sphinx/connector/delta-lake.rst [hunk 8]
```diff
@@ -6,24 +6,27 @@ Delta Lake connector
 
   <img src="../_static/img/delta-lake.png" class="connector-logo">
 
-The Delta Lake connector allows querying data stored in `Delta Lake
+The Delta Lake connector allows querying data stored in the `Delta Lake
 <https://delta.io>`_ format, including `Databricks Delta Lake
-<https://docs.databricks.com/delta/index.html>`_. It can natively read the Delta
-transaction log, and thus detect when external systems change data.
+<https://docs.databricks.com/delta/index.html>`_. The connector can natively
+read the Delta transaction log and thus detect when external systems change
```
Comment (mosabua) [inline docs/src/main/sphinx/connector/delta-lake.rst:12]: Please check if its called the "Delta transaction log" or the "Delta Lake transaction log"

File: docs/src/main/sphinx/connector/delta-lake.rst [hunk 9]
```diff
@@ -6,24 +6,27 @@ Delta Lake connector
...
+* Tables written by Databricks Runtime 7.3 LTS, 9.1 LTS, 10.4 LTS, 11.3 LTS, and
+  12.2 LTS are supported.
 * Deployments using AWS, HDFS, Azure Storage, and Google Cloud Storage (GCS) are
   fully supported.
 * Network access from the coordinator and workers to the Delta Lake storage.
 * Access to the Hive metastore service (HMS) of Delta Lake or a separate HMS.
 * Network access to the HMS from the coordinator and workers. Port 9083 is the
   default port for the Thrift protocol used by the HMS.
 
+
```
Comment (mosabua) [inline docs/src/main/sphinx/connector/delta-lake.rst:29]: Remove redundant empty line

File: docs/src/main/sphinx/connector/delta-lake.rst [hunk 10]
```diff
@@ -452,12 +459,12 @@ Flush metadata cache
...
 * ``system.flush_metadata_cache()``
 
-  Flush all metadata caches.
+  Flushes all metadata caches.
 
 * ``system.flush_metadata_cache(schema_name => ..., table_name => ...)``
 
-  Flush metadata caches entries connected with selected table.
-  Procedure requires named parameters to be passed
+  Flushes metadata cache entries connected with selected table.
```
Comment (mosabua) [inline docs/src/main/sphinx/connector/delta-lake.rst:466]:
```suggestion
  Flushes metadata cache entries connected with a specific table.
```

or 

```suggestion
  Flushes metadata cache entries of a specific table.
```

File: docs/src/main/sphinx/connector/delta-lake.rst [hunk 11]
```diff
@@ -869,7 +874,7 @@ connector.
 
    Performance tuning configuration properties are considered expert-level
    features. Altering these properties from their default values is likely to
-   cause instability and performance degradation. We strongly suggest that
+   cause instability and performance degradation. It is strongly suggested that
```
Comment (mosabua) [inline docs/src/main/sphinx/connector/delta-lake.rst:877]: I don't think we have a good standard for this .. maybe look up on google style .. but the passive voice here is probably not an improvement

**Review by mosabua — CHANGES_REQUESTED** [review]
A couple of more edits needed. Also please change the commit message to 

Improve style of Delta Lake connector documentation

File: docs/src/main/sphinx/connector/delta-lake.rst (hunk 8 above)
Comment (tlblessing) [reply docs/src/main/sphinx/connector/delta-lake.rst:12]: Looks like it should be Delta Lake transaction log https://www.databricks.com/blog/2019/08/21/diving-into-delta-lake-unpacking-the-transaction-log.html

Comment (tlblessing) [reply docs/src/main/sphinx/connector/delta-lake.rst:12]: My only original change was to go from the blind subject reference "It" to "The connector".

File: docs/src/main/sphinx/connector/delta-lake.rst (hunk 10 above)
Comment (tlblessing) [reply docs/src/main/sphinx/connector/delta-lake.rst:466]: Went with 2nd suggestion. Would just commit suggestion but would rather not rely on gui for Trino fork.

File: docs/src/main/sphinx/connector/delta-lake.rst (hunk 11 above)
Comment (tlblessing) [reply docs/src/main/sphinx/connector/delta-lake.rst:877]: Will look up "we" in google style but "we" implies that the reader is also the subject, which they're not.

Comment (tlblessing) [reply docs/src/main/sphinx/connector/delta-lake.rst:877]: No "we" https://developers.google.com/style/person

File: docs/src/main/sphinx/connector/delta-lake.rst [hunk 12]
```diff
@@ -452,12 +459,12 @@ Flush metadata cache
...
 
-  Flush all metadata caches.
+  Flushes all metadata caches.
 
 * ``system.flush_metadata_cache(schema_name => ..., table_name => ...)``
 
-  Flush metadata caches entries connected with selected table.
-  Procedure requires named parameters to be passed
+  Flushes metadata cache entries connected with selected table.
+  Procedure requires passing named parameters.
```
[the remark this answers is not in the document]
Comment (tlblessing) [reply docs/src/main/sphinx/connector/delta-lake.rst:467]: I was just editing the passive voice "to be". I did not change or embellish the material or change its meaning. This comment is scope creep. Agree the sentence does need improvement with technical details and less telegraph style.

File: docs/src/main/sphinx/connector/delta-lake.rst (hunk 11 above)
Comment (mosabua) [reply docs/src/main/sphinx/connector/delta-lake.rst:876]: Yeah... I know.. I dont like "we" and I know its wrong. Just that the passive voice is also kinda bad..
